docs/openapi/definitions/auth.yaml

```diff
@@ -0,0 +1,217 @@
+  - url: http://localhost:7000/auth/google
+tags:
+  - name: provider
+    description: List of endpoints per provider
+paths:
+  /start:
+    get:
+      tags:
+        - provider
+      summary:
+        Initializes the authorization flow and redirects to a consent popup for
+        the provider
+      description:
+        The popup window is pointed to the `/start` endpoint of the auth
+        provider in the auth-backend plugin, which then redirects to the OAuth
+        consent screen of the provider.
+      parameters:
+        - name: scope
+          in: query
+          description: Scope requested for the auth session
+          required: true
+          schema:
+            type: string
+            example: email
+        - name: env
+          in: query
+          description: Parameter to indicate runtime environment
+          required: true
+          schema:
+            type: string
+            example: development
+      responses:
+        302:
+          description: Redirect to the consent popup
+          headers:
+            Location:
+              description: The url to the consent popup
+              schema:
+                type: string
+            Set-Cookie:
+              description: Nonce cookie
+              schema:
+                $ref: '#/components/schemas/Nonce-Cookie'
+        default:
+          description: An error occurred
+  /frame/handler:
+    get:
+      tags:
+        - provider
+      summary: Handles the response from the consent popup for the provider
+      description:
+        If the login request is accepted, the popup window will be redirected
+        back to the `/handler/frame` endpoint of the auth backend.
+      parameters:
+        - name: scope
+          in: query
+          description: Scope requested for the auth session
+          required: true
+          schema:
+            type: string
+            example: email
+        - name: code
+          in: query
+          description:
+            Short-term authorization code used to be exchanged for long-term
+            tokens
+          required: true
+          schema:
+            type: string
+        - name: state
+          in: query
+          description: Nonce value stored in state
+          required: true
+          schema:
+            type: string
+        - name: provider-nonce
+          in: cookie
+          description: Nonce value stored in cookie
+          required: true
+          schema:
+            type: string
+        - name: env
+          in: query
+          description: Parameter to indicate runtime environment
+          required: true
+          schema:
+            type: string
+            example: development
+      responses:
+        200:
+          description: Message received from consent popup
+          headers:
+            Set-Cookie:
+              description: Refresh cookie
+              schema:
+                $ref: '#/components/schemas/Refresh-Cookie'
+    post:
```

Comment:
should this block be removed?

Reply:
Yeah, apparently this `post` is for SAML and this spec ended up being very `oauth` specific. Let's remove for now and revisit.